### PR TITLE
Use latest cabal master for haddocks generation. Fix broken haddocks.

### PR DESCRIPTION
**Pull request by carbolymer**
# Changelog

```yaml
- description: |
    Use latest cabal master for haddocks generation. Fix broken haddocks.
# uncomment types applicable to the change:
  type:
  # - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
  # - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  # - refactoring    # QoL changes
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
   - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

Use latest cabal master for haddocks generation. 
Fixes https://github.com/IntersectMBO/cardano-api/issues/437 
Fixes https://github.com/IntersectMBO/cardano-api/issues/546

Sample workflow run with generated haddocks: https://github.com/IntersectMBO/cardano-api/actions/runs/9778593657

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Self-reviewed the diff

### CODE REVIEW

File: .github/workflows/github-page.yml [hunk 1]
```diff
@@ -43,7 +43,7 @@ jobs:
     - name: Build documentation
       run: |
         nix develop .\#haddockShell --command \
-          cabal haddock-project --local --output=./haddocks --internal --foreign-libraries
```
Comment (prednaz) [inline .github/workflows/github-page.yml:46]: why remove `--local`?

**Comment by smelc** [comment]
@carbolymer> can we test the behavior before merging? (building locally?)

**Comment by carbolymer** [comment]
@smelc I've enabled for the development the workflow on a branch, and you can see the results there: https://github.com/IntersectMBO/cardano-api/actions/runs/9778593657
There's an artifact which contains generated haddocks.

File: .github/workflows/github-page.yml (hunk 1 above)
Comment (carbolymer) [reply .github/workflows/github-page.yml:46]: This command line argument is not present anymore in the latest cabal master.

**Review by smelc — APPROVED** [review]

**Comment by carbolymer** [comment]
> github-merge-queue bot removed this pull request from the merge queue due to no response for status checks 
CI is green, so force merging.

**Pull request merged**